src/lib.rs

```diff
@@ -110,12 +123,13 @@ where
         // assert that it was droppeds
         unsafe { ptr::drop_in_place(next) };
         loop {
-            let pcurrent = self.inner.load(load_order);
+            let pcurrent = self.inner.load(Ordering::Acquire);
             let current = unsafe { Self::inner_from_raw(pcurrent) };
             unsafe { ptr::write(next, current) };
-            let last = self.inner.compare_and_swap(pcurrent, raw, cas_order);
-            if last == pcurrent {
-                return last.is_null();
+            let result = self.inner.compare_exchange(pcurrent, raw, Ordering::AcqRel, Ordering::Acquire);
```

Comment:
The returned pointer is never dereferenced, so the `Acquire` ordering is redundant here.

```suggestion
            let result = self.inner.compare_exchange(pcurrent, raw, Ordering::Release, Ordering::Relaxed);
```